src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -101,26 +100,18 @@ public function getParameters(): array
                 $paramNode,
                 $this,
                 $paramIndex,
-                $this->isParameterOptional($nodeParams, $paramNode, $paramIndex),
+                $this->isParameterOptional($nodeParams, $paramIndex),
             );
         }
 
         return $parameters;
     }
 
     /** @param list<Node\Param> $parameterNodes */
-    private function isParameterOptional(array $parameterNodes, ParamNode $parameterNode, int $parameterIndex): bool
+    private function isParameterOptional(array $parameterNodes, int $parameterIndex): bool
```

Comment:
Somehow, `roave/backward-compatibility-check` [picked up this change](https://github.com/Roave/BetterReflection/actions/runs/3111377883/jobs/5043591213):

```
[BC] CHANGED: The parameter $parameterNode of Roave\BetterReflection\Reflection\ReflectionFunctionAbstract#isParameterOptional() changed from PhpParser\Node\Param to a non-contravariant int
[BC] CHANGED: The parameter $parameterNode of Roave\BetterReflection\Reflection\ReflectionFunctionAbstract#isParameterOptional() changed from PhpParser\Node\Param to int
[BC] CHANGED: Parameter 1 of Roave\BetterReflection\Reflection\ReflectionFunctionAbstract#isParameterOptional() changed name from parameterNode to parameterIndex
```

Ignoring for now, just an interesting bug though: possibly due to `abstract` class not having many filters on method visibility analysis.